src/Console/ConfigurationResolver.php

```diff
@@ -702,16 +703,106 @@ private function validateRules(array $rules): void
+            	],
+                'no_multiline_whitespace_before_semicolons' =>
+            	[
+            		'new_name' => 'multiline_whitespace_before_semicolons',
+            		'note' => []
+            	],
+                'no_short_echo_tag' =>
+            	[
+            		'new_name' => 'echo_tag_syntax',
+            		'note' => ['format' => 'long']    
+            	],
+                'php_unit_ordered_covers' =>
+            	[
+            		'new_name' => 'phpdoc_order_by_value',
+            		'note' => ['annotations' => ['covers']] 
+            	],
+                'phpdoc_inline_tag' =>
+            	[
+            		'new_name' => 'general_phpdoc_tag_rename, phpdoc_inline_tag_normalizer and phpdoc_tag_type',
+            		'note' => []    
+            	],
+                'pre_increment' =>
+            	[
+            		'new_name' => 'increment_style',
+            		'note' => ['style' => 'pre']    
+            	],
+                'psr0' =>
+            	[
+            		'new_name' => 'psr_autoloading',
+            		'note' => ['dir' => 'x'],    
+            	],
+                'psr4' =>
+            	[
+            		'new_name' => 'psr_autoloading',
+            		'note' => ''    
+            	],
+                'silenced_deprecation_error' =>
+            	[
+            		'new_name' => 'error_suppression',
+            		'note' => ''    
+            	],
+                'trailing_comma_in_multiline_array',
+            	[
+            		'new_name' => 'trailing_comma_in_multiline',
+            		'note' => ['elements' => ['arrays']],  
+            	]
+            ];
+            
             $matcher = new WordMatcher($availableFixers);
 
             $message = 'The rules contain unknown fixers: ';
             foreach ($unknownFixers as $unknownFixer) {
-                $alternative = $matcher->match($unknownFixer);
-                $message .= sprintf(
-                    '"%s"%s, ',
-                    $unknownFixer,
-                    null === $alternative ? '' : ' (did you mean "'.$alternative.'"?)'
-                );
+
+                $isRenamed = false;
+                // Check if present as old renamed rule
+                if(isset($renamedRules[$unknownFixer])) {
+                     $message .= sprintf(
+                        '"%s" is a renamed rule, %s, ',
+                        $unknownFixer,
+                        ' did you mean "'.$renamedRules[$unknownFixer]['new_name'].'"?' . (!empty($renamedRules[$unknownFixer]['note']) ? ' note: use configuration ' . HelpCommand::toString($renamedRules[$unknownFixer]['note']) : '' ) . PHP_EOL . PHP_EOL .
+                        'For more info see: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/UPGRADE-v3.md#renamed-rules'
+                    );
+                         
+                    $isRenamed = true;
+                }
+
+                // Go to normal matcher if it is not a renamed rule
+                if (!$isRenamed) {
```

Comment:
we do not need `isRenamed` variable.
replace it with `if { ... } else { ... }`

Reply:
Done in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5887/commits/cdd1b3d6329fac933a3249778a4211a164ef76a0